yourgov/components/FormMobileFoodVendorPermit/steps/FormStep9.tsx

```diff
@@ -1,72 +1,292 @@
+import { SectionAlert } from '@ag.ds-next/react/section-alert';
+import { FileUpload, FileWithStatus } from '@ag.ds-next/react/file-upload';
+import { Text } from '@ag.ds-next/react/text';
+import { Box } from '@ag.ds-next/react/box';
+import { H2 } from '@ag.ds-next/react/heading';
+import { visuallyHiddenStyles } from '@ag.ds-next/react/a11y';
 import { useGlobalForm } from '../GlobalFormProvider';
-import { FormContainer } from './FormContainer';
+import { StepActions } from '../StepActions';
+import { UploadFileTable } from '../UploadFileTable';
 import { useFormContext } from './FormProvider';
-import { step9FormSchema, type Step9FormSchema } from './FormState';
+import { FormContainer } from './FormContainer';
+
+const tableHeadingId = 'upload-documents-heading';
+const tableId = 'upload-file-table';
+
+const documentRows = [
+	{
+		documentType: 'RMS Vehicle registration',
+		file: '',
+		size: '',
+		id: 'rms-vehicle-registration',
+		error: false,
+	},
+	{
+		documentType: 'Food Safety Supervisor Certificate for Charlie Walker',
+		file: '',
+		size: '',
+		id: 'food-safety-certificate',
+		error: false,
+	},
+	{
+		documentType: 'Suggested menu or list of foods being sold',
+		file: '',
+		size: '',
+		id: 'suggested-menu-or-list-of-foods-being-sold',
+		error: false,
+	},
+] as const;
+
+export type Document = Pick<
+	(typeof documentRows)[number],
+	'documentType' | 'id'
+> & {
+	file: string;
+	size: string;
+	error: boolean;
+};
+
+type SampleTableError = {
+	documentType: Document['documentType'];
+	href: string;
+	message: string;
+	name: string;
+};
 
 export function FormStep9() {
-	const { formState, step9SetState, isSavingBeforeExiting } = useGlobalForm();
+	const { step9GetState, step9SetState, isSavingBeforeExiting } =
+		useGlobalForm();
 	const { submitStep } = useFormContext();
 
-	const {
-		control,
-		handleSubmit,
-		formState: { errors },
-	} = useForm<Step9FormSchema>({
-		defaultValues: formState.steps?.step9,
-		resolver: isSavingBeforeExiting ? undefined : zodResolver(step9FormSchema),
-		mode: 'onSubmit',
-		reValidateMode: 'onBlur',
-	});
+	const errorMessageRef = useRef<HTMLDivElement>(null);
+	const successMessageRef = useRef<HTMLDivElement>(null);
+
+	const { files = {} } = step9GetState() || {};
+
+	// Table data
+	const [documents, setDocuments] = useState<Document[]>(
+		documentRows.map((documentRow) => ({
+			...documentRow,
+			file: files?.[documentRow.id]?.file || '',
+			size: files?.[documentRow.id]?.size || '',
+		}))
+	);
+	const [currentDocument, setCurrentDocument] = useState<Document>();
+
+	// Errors
+	const [tableErrors, setTableErrors] = useState<SampleTableError[]>([]);
+
+	// Drawer state
+	const [isDrawerOpen, openDrawer, closeDrawer] = useTernaryState(false);
+	const [uploadedFile, setUploadedFile] = useState<FileWithStatus[]>([]);
+
+	// Action handlers
+	const closeDrawerAndClearForm = () => {
+		setUploadedFile([]);
+		closeDrawer();
+	};
+
+	const uploadFile = () => {
```

Comment:
I think if you press this but haven't picked a filen then we should mark it as invalid and focus it

Reply:
Right now it incorrectly adds the success banner even though there was no file

Reply:
- [ ] invalid row when no file is chosen